modules/contact/test/tests/mortar_aux_kernels/pressure-aux-friction-3d.i

```diff
@@ -0,0 +1,372 @@
+    variable = tangent2_lm
+    primary_boundary = 'bottom_top'
+    secondary_boundary = 'top_bottom'
+    lm_var_x = lm_x
+    lm_var_y = lm_y
+    lm_var_z = lm_z
+    component = 'tangent2'
+    boundary = 'top_bottom'
+  []
+[]
+
+[Constraints]
+  [weighted_gap_lm]
+    type = ComputeFrictionalForceCartesianLMMechanicalContact
+    primary_boundary = 'bottom_top'
+    secondary_boundary = 'top_bottom'
+    primary_subdomain = 'primary_lower'
+    secondary_subdomain = 'secondary_lower'
+    lm_x = lm_x
+    lm_y = lm_y
+    lm_z = lm_z
+    variable = lm_x # This can be anything really
+    disp_x = disp_x
+    disp_y = disp_y
+    disp_z = disp_z
+    use_displaced_mesh = true
+    correct_edge_dropping = true
+    c = 1e+02
+    c_t = 1e+2
+    mu = 0.10
+  []
+  [normal_x]
+    type = CartesianMortarMechanicalContact
+    primary_boundary = 'bottom_top'
+    secondary_boundary = 'top_bottom'
+    primary_subdomain = 'primary_lower'
+    secondary_subdomain = 'secondary_lower'
+    variable = lm_x
+    secondary_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+    correct_edge_dropping = true
+  []
+  [normal_y]
+    type = CartesianMortarMechanicalContact
+    primary_boundary = 'bottom_top'
+    secondary_boundary = 'top_bottom'
+    primary_subdomain = 'primary_lower'
+    secondary_subdomain = 'secondary_lower'
+    variable = lm_y
+    secondary_variable = disp_y
+    component = y
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+    correct_edge_dropping = true
+  []
+  [normal_z]
+    type = CartesianMortarMechanicalContact
+    primary_boundary = 'bottom_top'
+    secondary_boundary = 'top_bottom'
+    primary_subdomain = 'primary_lower'
+    secondary_subdomain = 'secondary_lower'
+    variable = lm_z
+    secondary_variable = disp_z
+    component = z
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+    correct_edge_dropping = true
+  []
+[]
+
+[BCs]
+  [botx]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 'bottom_left bottom_right bottom_front bottom_back'
+    value = 0.0
+  []
+  [boty]
+    type = DirichletBC
+    variable = disp_y
+    boundary = 'bottom_left bottom_right bottom_front bottom_back'
+    value = 0.0
+  []
+  [botz]
+    type = DirichletBC
+    variable = disp_z
+    boundary = 'bottom_left bottom_right bottom_front bottom_back'
+    value = 0.0
+  []
+  [topx]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 'top_top'
+    value = 0.0
+  []
+  [topy]
+    type = DirichletBC
+    variable = disp_y
+    boundary = 'top_top'
+    value = 0.0
+  []
+  [topz]
+    type = FunctionDirichletBC
+    variable = disp_z
+    boundary = 'top_top'
+    function = '-${starting_point} * sin(2 * pi / 40 * t) + ${offset}'
+  []
+[]
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+  []
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 1
+  dt = .5
+  dtmin = .01
+  solve_type = 'NEWTON'
```

Comment:
Newton? Does it do better than PJFNK on this problem? If we're using Newton, then we can remove `-mat_mffd_err` from the petsc options

Reply:
Similarly for any other inputs with Newton and `-mat_mffd_err`

Reply:
Many mechanical problems with mortar contact can do as well with Newton as with PJFNK. Not all of them, but many. 
I think I removed all unused `-mat_mffd_err`.

Also, the variable condensation preconditioner has shown to work best for Newton compared to PJFNK, this is to be investigated.